lib/ash/data_layer/ets/ets.ex

```
@@ -1653,6 +1642,21 @@ defmodule Ash.DataLayer.Ets do
     end
   end
 
+  defp get_valid_pkey(resource, changeset) do
+    pkey =
+      resource
+      |> Ash.Resource.Info.primary_key()
+      |> Enum.into(%{}, fn attr ->
+        {attr, Ash.Changeset.get_attribute(changeset, attr)}
+      end)
+
+    if !Enum.empty?(pkey) && Enum.all?(pkey, fn {_, v} -> is_nil(v) end) do
```

If any primary key is `nil` here we can raise an error.